docs/source/reference/cuda.rst

```diff
@@ -35,6 +35,46 @@ Memory management
    cupy.cuda.PinnedMemoryPool
 
 
+CuPy provides C API to access its current device memory allocator. To use this,
+please include ``cupy/cuda/cupy_memory.h`` in your C/C++ code.
+
+.. c:function:: cupy_allocator_handle* get_cupy_allocator_handle()
+
+    This function creates an opaque handle to CuPy's device memory allocator.
+
+    Internally, it inspects if the Python interpreter is already initialized
+    in the current thread, calls ``Py_Initialize()`` if not, and import the
+    necessary functions from :mod:`cupy.cuda.memory` for use.
+
+.. c:function:: void destroy_cupy_allocator_handle(cupy_allocator_handle* ptr)
+
+    This function destroys the handle ``ptr`` to CuPy's device memory allocator.
+    Any device memory allocated from the handle should be freed before destroying
+    it.
+
+    Internally, it checks whether ``Py_Initialize()`` was called when the handle
+    was created, and calls ``Py_Finalize()`` if so.
+
+.. c:function:: void* cupy_malloc(cupy_allocator_handle* handle, size_t size)
+
+    This function returns a device pointer to the memory of ``size`` bytes
```

Comment:
Alignment?

Reply:
CuPy hard-codes it to 512 bytes IIUC: https://github.com/cupy/cupy/blob/ff18a56f23058a947e5ceab74995deda8514b2f1/cupy/cuda/memory.pyx#L679-L681

Reply:
Note that `cudaMalloc` is aligned to (at least) 256B, not 512.

I think it should just be documented here as well.

Reply:
Jake, thanks for the inside info! I don't think this is ever documented publicly, is it? I also spent some time trying to verify this a while ago but in vain, so I came up [a theory](https://gist.github.com/sonots/41daaa6432b1c8b27ef782cd14064269#gistcomment-3014146) for why this is the case...

Reply:
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#device-memory-accesses

> Any address of a variable residing in global memory or returned by one of the memory allocation routines from the driver or runtime API is always aligned to at least 256 bytes.

Reply:
Jake, are you able to verify this comment? https://gist.github.com/sonots/41daaa6432b1c8b27ef782cd14064269#gistcomment-2127860

Reply:
The requirement is for _at least_ 256B. It is unsafe to rely on it being anything larger.